src/components/Referrals/ReferralJourney/ReferralJourney.tsx

```diff
@@ -18,22 +18,40 @@ import {
   JourneyContainer,
   JourneyWrapper
 } from './ReferralJourney.styled'
-import { ANIMATION_DURATION } from './utils'
+import { ANIMATION_DURATION, getTiersCompletedCount } from './utils'
 import { AnimationPhase as AnimationPhase, ReferralJourneyProps } from './ReferralJourney.types'
 
 // eslint-disable-next-line import/no-named-as-default-member
 const ReferralJourney = React.memo((props: ReferralJourneyProps) => {
-  const { invitedUsersAccepted, onSetReferralEmail } = props
+  const { invitedUsersAccepted, invitedUsersAcceptedViewed, onSetReferralEmail, rewardImages } = props
+
   const [animatedStep, setAnimatedStep] = useState(TIERS.filter(tier => tier.invitesAccepted <= invitedUsersAccepted).length)
   const [open, setOpen] = useState(false)
   const [journeyTiers, setJourneyTiers] = useState(
     TIERS.map(tier => ({ ...tier, completed: tier.invitesAccepted <= invitedUsersAccepted }))
   )
   const [animationPhase, setAnimationPhase] = useState<AnimationPhase>(AnimationPhase.WAITING_NEXT_TIER)
+  const [currentTierIndex, setCurrentTierIndex] = useState<number>(-1)
   const totalSteps = TIERS.length
   const journeyStepRefs = useRef<(Element | null)[]>([])
   const sectionRef = useRef<HTMLDivElement>(null)
 
+  const getNextTierIndex = useCallback((): number => {
+    return TIERS.findIndex((tier, index) => {
+      const isCurrentlyCompleted = tier.invitesAccepted <= invitedUsersAccepted
+      const wasCompletedWhenViewed = tier.invitesAccepted <= invitedUsersAcceptedViewed
+      const isNewlyCompleted = isCurrentlyCompleted && !wasCompletedWhenViewed
+
+      return isNewlyCompleted && index > currentTierIndex
+    })
```

Comment:
Can this be computed with `getNewTiersToAnimate`?
```suggestion
  const newTiersToAnimate = getNewTiersToAnimate()
  return tiers.length + 1 - getNewTiersToAnimate()
```
We would need to change how `-1` is treated.

Reply:
You can't use getNewTiersToAnimate alone to determine the next tier index, because it only tells you how many new tiers are completed, not which specific tier should be animated next; you still need to find the first newly completed tier’s index to animate them one by one.